ztp/ztp-policy-generator/kustomize/plugin/policyGenerator/v1/policygenerator/policyGen/policyBuilder.go

```diff
@@ -37,39 +41,50 @@ func (pbuilder *PolicyBuilder) Build(policyGenTemp utils.PolicyGenTemplate) (map
 			if err != nil {
 				return policies, err
 			}
-			name := policyGenTemp.Metadata.Name + "-" + sFile.PolicyName
-			path := policyGenTemp.Metadata.Name + "/" + policyGenTemp.Metadata.Name + "-" + sFile.PolicyName
 			if sFile.PolicyName == "" {
-				// Generate CR without append to policy
+				// Generate plain CRs (no policy)
 				if len(resources) > 1 {
-					return policies, errors.New("Update spec/data of multiple yamls structure in same file " + sFile.FileName +
-						" not allowed. Instead separate them in multiple files")
-				}
-				resource := resources[0]
-				name := resource["kind"].(string)
-				name = name + "-" + resource["metadata"].(map[string]interface{})["name"].(string)
-
-				if resource["metadata"].(map[string]interface{})["namespace"] != nil {
-					name = name + "-" + resource["metadata"].(map[string]interface{})["namespace"].(string)
-				}
-				policies[utils.CustomResource+"/"+policyGenTemp.Metadata.Name+"/"+name] = resource
-			} else if sFile.PolicyName != "" && policies[path] == nil {
-				// Generate new policy
-				acmPolicy, err := pbuilder.createAcmPolicy(name, policyGenTemp.Metadata.Namespace, resources)
-				if err != nil {
-					return policies, err
+					// Multi-document yaml - use the source filename
+					name := fmt.Sprintf("Multiple-%s", strings.TrimSuffix(sFile.FileName, filepath.Ext(sFile.FileName)))
+					output := path.Join(utils.CustomResource, policyGenTemp.Metadata.Name, name)
+					policies[output] = resources
+				} else {
+					// Single-resource yaml - construct a filename based on the contents
+					resource := resources[0]
+					nameParts := []string{
+						resource["kind"].(string),
+						resource["metadata"].(map[string]interface{})["name"].(string),
+					}
+					resourceNamespace := resource["metadata"].(map[string]interface{})["namespace"]
+					if resourceNamespace != nil {
+						nameParts = append(nameParts, resourceNamespace.(string))
+					}
+					name := strings.Join(nameParts, "-")
+					output := path.Join(utils.CustomResource, policyGenTemp.Metadata.Name, name)
+					policies[output] = resource
```

Comment:
Minor nit but `policies[output] = resources` could go outside the if/else block to show that the difference is only in the name (output) and not in the content of 'resources'.

Reply:
I suppose - The difference is that right now we pass a single map[string]string if it's a single resource and a slice []map[string]string if there's more than one.... But I guess at this level we could always pass the whole list whether it has 1 or more...

Reply:
This was a good idea - I think the new version I just pushed is a lot cleaner because of it.